alerting/src/main/kotlin/org/opensearch/alerting/settings/AlertingSettings.kt

```diff
@@ -156,5 +173,94 @@ class AlertingSettings {
+            DEFAULT_TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS,
+            -1, TotalMaxActionsTriggersValidator(internalClient),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+        )
+
+        internal class TotalMaxActionsTriggersValidator(val client: Client?) : Setting.Validator<Int> {
+            override fun validate(value: Int) {}
+
+            override fun validate(value: Int, settings: Map<Setting<*>, Any>) {
+                val maxActions = settings[TOTAL_MAX_ACTIONS_PER_TRIGGERS] as Int
+                validateActionsTrigger(maxActions, value, client)
+            }
+
+            override fun settings(): MutableIterator<Setting<*>> {
+                val settings = mutableListOf<Setting<*>>(
+                    TOTAL_MAX_ACTIONS_PER_TRIGGERS
+                )
+                return settings.iterator()
+            }
+        }
+
+        internal class MaxActionsTriggersValidator(val client: Client?) : Setting.Validator<Int> {
+            override fun validate(value: Int) {}
+
+            override fun validate(value: Int, settings: Map<Setting<*>, Any>) {
+                val totalMaxActions = settings[TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS] as Int
+                validateActionsTrigger(value, totalMaxActions, client)
+            }
+
+            override fun settings(): MutableIterator<Setting<*>> {
+                val settings = mutableListOf<Setting<*>>(
+                    TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS
+                )
+                return settings.iterator()
+            }
+        }
+
+        private fun validateActionsTrigger(maxActions: Int, totalMaxActions: Int, client: Client?) {
+            if (maxActions > totalMaxActions) {
+                throw IllegalArgumentException(
+                    "The limit number of actions for a single trigger, $maxActions, " +
+                        "should not be greater than that of the overall max actions across all triggers of the monitor, $totalMaxActions"
+                )
+            }
+            client?.let {
+                GlobalScope.launch {
+                    val triggers = getTriggers(it)
+
+                    var currentAmountOfActions = getCurrentAmountOfActions(triggers)
+                    currentAmountOfActions += maxActions
+
+                    if (currentAmountOfActions > totalMaxActions)
+                        throw IllegalArgumentException(
+                            "The amount of actions that the client wants to update plus the amount of actions that " +
+                                "already exist, $currentAmountOfActions should not be greater than  that of the " +
+                                "overall max actions across all triggers of the monitor, $totalMaxActions"
+                        )
+                }
+            }
+        }
+
+        suspend fun getTriggers(client: Client): List<Trigger> {
+            val getMonitorRequest = GetMonitorRequest("", 1L, RestRequest.Method.GET, null)
```

Comment:
This returns 0 monitors. We want to do a searchMonitor call to get all the existing monitors and get their triggers.